Pomodoro/app/src/main/java/no/uia/ikt205/pomodoro/MainActivity.kt

```diff
@@ -35,31 +50,79 @@ class MainActivity : AppCompatActivity(), View.OnClickListener {
                stopCountDown()
        }
 
-        // Find countdown buttons and set onClick listeners
-        val buttonList = arrayOf(btn30min, btn60min, btn90min, btn120min)
-        buttonList.forEach { it.setOnClickListener(this) }
+        repetitionCounter = findViewById(R.id.repetitionCounter)
 
-       coutdownDisplay = findViewById(R.id.countDownView)
+        workSeekBar = findViewById(R.id.workSeekBar)
+        workTimer = findViewById(R.id.workTimer)
+        workSeekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
+            override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
+                timeToCountDownInMinWorkTime = progress + minWorkTime // Calculate the work time from seekbar
+                updateSeekBar(timeToCountDownInMinWorkTime, workTimer)
+            }
+
+            override fun onStartTrackingTouch(seekBar: SeekBar?) {
+                //TODO("Not yet implemented")
+            }
+
+            override fun onStopTrackingTouch(seekBar: SeekBar?) {
+                //TODO("Not yet implemented")
+            }
+        })
+        updateSeekBar(timeToCountDownInMinWorkTime, workTimer)
+
+        pauseSeekbar = findViewById(R.id.pauseSeekBar)
+        pauseTimer = findViewById(R.id.pauseTimer)
+        pauseSeekbar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
+            override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
+                timeToCountDownInMinPauseTime = progress + minPauseTime // Calculate the pause time from seekbar
+               updateSeekBar(timeToCountDownInMinPauseTime, pauseTimer)
+            }
 
+            override fun onStartTrackingTouch(seekBar: SeekBar?) {
+                //TODO("Not yet implemented")
+            }
+
+            override fun onStopTrackingTouch(seekBar: SeekBar?) {
+                //TODO("Not yet implemented")
+            }
+        })
+        updateSeekBar(timeToCountDownInMinPauseTime, pauseTimer)
     }
 
+
     // Starts the countdown
     private fun startCountDown() {
-
         timer = object : CountDownTimer(timeToCountDownInMs,timeTicks) {
             override fun onFinish() {
-                Toast.makeText(this@MainActivity,"Arbeidsøkt er ferdig", Toast.LENGTH_SHORT).show()
                 startButton.text = getString(R.string.startButtonText)
                 cancel() // Stop timer
+                work = !work // Flip timer status between work and pause
+                if (getRepetitionCounter() > 1) {
+                    decrementRepetitionCounter()
+                    updateSeekBar(timeToCountDownInMinWorkTime, workTimer)
+                    updateSeekBar(timeToCountDownInMinPauseTime, pauseTimer)
+                    startCountDown()
```

Comment:
Er det meningen at den skal kalle seg selv? Ser ikke helt hvorfor det skal være nødvendig

Reply:
Ja det er nødvendig, for å starte countdown på nytt. Eventuelt kan jeg lage en ny countDownTimer object og ha en countdowntimer for pause og en for work, men følte det var unødvendig ettersom den skal gjøre det samme.